### ISSUE TITLE
Help with initialising a job that requires many dependencies

### ISSUE THREAD

**Issue by tomaytotomato** [spinscale/dropwizard-jobs#70]
Hi all,

Apologies but this question is a little out of scope but I am hoping you can help me.

Currently in Dropwizard I am manually triggering a job through an endpoint to update records which don't have a UUID value.

I want to now make this a Quartz job however I am having issues with initialising my dependencies which require `Configuration` instead of `Bootstrap` , as they were originally managed in Dropwizard's `run()` method.

Looking for advice on how to implement. I've made sure my App config implements `JobConfiguration`

This is the code here which uses JDBI and Jersey HTTP client, which is not working.

I tried using bootstrap.getConfiguration

```
        final Client client = new JerseyClientBuilder(environment).using(config.getJerseyClientConfiguration())
            .build(getName());
        
        final ReportClient reportClient = new ReportClientJerseyImpl(client, config.getReportProxy().getTargetUri());
        
        final ShepherdReportUUIDJob reportUUIDJob = new ShepherdReportUUIDJob(jdbi,reportClient);
        bootstrap.addBundle(new JobsBundle(reportUUIDJob));

```

So in summary, what do I need to do to get my dependencies for the Quartz job to work in `initialise()` rather than `run()`?

**Comment by bpicolo** [comment]
Looks like some people are using juice, but I think just extending the jobs bundle kinda like e.g. hibernate and similar bundles do should do the trick. Here's what I'm thinking (not tested yet)

```java
package com.bundle;

import de.spinscale.dropwizard.jobs.Job;
import de.spinscale.dropwizard.jobs.JobManager;
import de.spinscale.dropwizard.jobs.JobsBundle;
import io.dropwizard.setup.Environment;
import com.YourConfigurationClass;

public class YourJobsBundle<T extends YourConfigurationClass> extends JobsBundle {

    protected JobManager jobManager;

    public void run(final T configuration, final Environment environment) throws Exception {
        jobManager = new JobManager(getJobs(configuration, environment));
        jobManager.configure(configuration);
        environment.lifecycle().manage(jobManager);
    }

    public Job[] getJobs(final T configuration, final Environment environment) {
        return new Job[]{};
    }
}

```

**Comment by hakandilek** [comment]
I'm going to close this one because it does not look like an issue about dwj.

**Comment by tomaytotomato** [comment]
Ok, I am still stuck with this as the job requires dependencies that need to access the config. 

Can I add my config values to the quartz and access them through that?
